connected_users/frame.py

```diff
@@ -0,0 +1,57 @@
+import psutil as ps
+import tkinter as tk
+from integrator import frame
+from logger.logger import Logger
+from datetime import datetime
+
+
+class frame(frame.DOSMFrame):
+    def __init__(self, master, logger: Logger, **options):
+        super().__init__(master, logger, **options)
+        self.logger = logger
+
+        self.columnconfigure(0, weight=1)
+        self.rowconfigure(0, weight=1)
+
+    def show(self):
+        # Define datagrid headings
+        columns = {
+            "name": "Name",
+            "host": "Host",
+            "started": "Login date",
+            "connexion_pid": "Connexion PID"
+        }
+
+        # Init datagrid
+        self.datagrid = tk.ttk.Treeview(
+            self, columns=list(columns.keys()), show='headings')
+
+        # Set datagrid headings
+        for column in columns.keys():
+            self.datagrid.heading(column, text=columns[column])
+
+        self.datagrid.pack(side=tk.LEFT, fill=tk.BOTH, expand=True)
+
+        # Init scrollbar
+        scrollbar = tk.ttk.Scrollbar(
+            self, orient=tk.VERTICAL, command=self.datagrid.yview)
+        self.datagrid.configure(yscroll=scrollbar.set)
+        scrollbar.pack(side=tk.RIGHT, fill=tk.Y)
+
+    def update(self, dt):
+        # Clear data grid
+        for item in self.datagrid.get_children():
+            self.datagrid.delete(item)
+
+        users = []
+        for user in ps.users():
+            users.append(
+                (user.name, user.host, datetime.fromtimestamp(user.started), user.pid))
+            self.logger.write_log(users[len(users) - 1])
```

Comment:
Suggestion par rapport à la proposition de @RedbeanGit : [la proposition en question](https://github.com/do3-2021/dosm/pull/20/files/3c1b6b1d9877045c20d9fff076ef7bbb5e7bfd68#r810615131)
```suggestion
            self.logger.write_log(users[-1])
```